pkg/callback/scheduler_callback.go

```diff
@@ -88,6 +88,27 @@ func (callback *AsyncRMCallback) RecvUpdateResponse(response *si.UpdateResponse)
 		}
 	}
 
+	// handle status changes
+	for _, updated := range response.UpdatedApplications {
+		log.Logger().Debug("status update callback received",
+			zap.String("appId", updated.ApplicationID),
+			zap.String("new status", updated.State))
+		// delete application from context
+		if updated.State == events.States().Application.Completed {
+			err := callback.context.RemoveApplicationInternal(updated.ApplicationID)
+			if err != nil {
+				log.Logger().Error("failed to delete application", zap.Error(err))
+			}
+		} else {
+			if updated.State == "Failing" || updated.State == events.States().Application.Failed {
+				ev := cache.NewFailApplicationEvent(updated.ApplicationID)
+				dispatcher.Dispatch(ev)
+			}
+			// handle status update
+			dispatcher.Dispatch(cache.NewApplicationStatusChangeEvent(updated.ApplicationID, events.AppStateChange, updated.State))
+		}
+	}
+
```

Comment:
I do not think this order matters, can we move it back?

Reply:
moved it back